CRM/Price/BAO/PriceField.php

```diff
@@ -291,26 +291,15 @@ public static function addQuickFormElement(
 
     //use value field.
     $valueFieldName = 'amount';
-    $separator = '|';
-    $taxTerm = Civi::settings()->get('tax_term');
-    $displayOpt = Civi::settings()->get('tax_display_settings');
-    $invoicing = Civi::settings()->get('invoicing');
     switch ($field->html_type) {
       case 'Text':
         $optionKey = key($customOption);
-        $count = $customOption[$optionKey]['count'] ?? '';
-        $max_value = $customOption[$optionKey]['max_value'] ?? '';
-        $taxAmount = $customOption[$optionKey]['tax_amount'] ?? NULL;
-        if (isset($taxAmount) && $taxAmount && $displayOpt && $invoicing) {
-          $qf->assign('displayOpt', $displayOpt);
-          $qf->assign('taxTerm', $taxTerm);
-          $qf->assign('invoicing', $invoicing);
-        }
-        $priceVal = implode($separator, [
-          $customOption[$optionKey][$valueFieldName] + $taxAmount,
-          $count,
-          $max_value,
-        ]);
+        // Text elements have a label before and then a second label after with amount, etc
+        // The second label is added here, we start by clearing out the existing text which is the first label.
+        $customOption[$optionKey]['label'] = NULL;
+        $priceOptionText = self::buildPriceOptionText($customOption[$optionKey], $field->is_display_amounts, $valueFieldName);
+        // This second label is then added to the form as a second form element which just carries the label and is not otherwise used.
+        $elementLabelAfter = &$qf->add('static', $elementName . '_label_after', $priceOptionText['label']);
```

Comment:
Minor thing: I'm guessing the `&$qf` was copied from somewhere. I'd be surprised if the `&` is needed.

Reply:
Yes, copied from elsewhere in this file. I don't think it is needed, have removed.